tests/test_transformation/test_arcsin_transformer.py

```diff
@@ -0,0 +1,74 @@
+import pandas as pd
+import pytest
+from sklearn.exceptions import NotFittedError
+
+from feature_engine.transformation import ArcsinTransformer
+
+
+def test_automatically_find_variables(df_vartypes):
```

Comment:
Finding variables automatically is one of the common tests that we run from our check_estimators tests. So I think we are good to exclude all bits to corroborating variables, leave only the part that tests the transformer output, and rename the test to something relevant. 


The variable tests we have in house are in [this file](https://github.com/feature-engine/feature_engine/blob/main/tests/estimator_checks/variable_selection_checks.py).

Reply:
I have a question (sorry if I ask naive questions but I am new to this kind of stuff). I do not understand where the data in df_vartypes are coming from. I mean when I run pytest on the test file I am executing the functions inside the file but I do not pass the value of df_vartypes (or df_na) so I do not understand where the table I can print (with Mark, Age ... ) comes from. Moreover I have also trouble locating the fit_transform method definition.

Reply:
Hey no problem at all! Happy to help. 

There is a [conf.py](https://github.com/feature-engine/feature_engine/blob/main/tests/conftest.py) file within the test module that contains a lot of dataframes that are used across tests. As you can see, the dataframes have a pytest..fixture at the top, that signals that those dfs can be used by tests across the code base.

In that conf.py file, we put dataframes that are used across test folders, for example test_encoding, test_transformation etc. Sometimes, the test folders also have a conf.py where we store dataframes that will be used solely by the tests within those folders.

When you mark dfs with a pytest fixture, then you can use them just by passing them to the test like we do, for example: test_something(df_vartypes). Pytest will know where to search for that dataframe.

Hope that helps :)

Reply:
Thanks! And the fit_transform method? Where do I find it?

Reply:
That one comes from scikit-learn's class `TransformerMixin`, which is inherited by the `BaseNumericalTransformer()` class, which is the parent class of the class you are creating here.

Reply:
So when I execute fit_transform(df_vartypes) the method in TransformerMixin is executed and this method call fit() and transform() methods the we wrote in the arcsin.py file. Is this correct?

Reply:
yes.